### PR TITLE
Rename @jupyterlab/server-proxy to @jupyterhub/jupyter-server-proxy

### LINKED ISSUE #363: Relocate jupyterlab extension from `@jupyterlab/server-proxy` to `@jupyterhub/jupyter-server-proxy`
We are now in control of the @jupyterhub NPM organization, and I figure we should relocate our extension from the NPM organization/package `@jupyterlab/server-proxy` to `@jupyterhub/jupyter-server-proxy`.

I figure this change should be considered breaking, and ship with 4.0.0 if so.

**Is there agreement to go for this?** If so, I'll work it!

## Related
- #323, closed as stale and insufficient
- #322, resolved by documentation

### PR DESCRIPTION
**Pull request by consideRatio**
- Closes #363 
- CI credentials: I've pushed 4.0.0-alpha.1 to [`@jupyterhub/jupyter-server-proxy`](https://www.npmjs.com/package/@jupyterhub/jupyter-server-proxy) and generated an access token.
- Renames the folder `jupyterlab-server-proxy` to `labextension` for readability now that the extension is named the same as the python package

### CODE REVIEW

**Review by manics — COMMENTED** [review]
How old is JupyterLab 2? Given this is a major release can we drop it?

File: RELEASE.md [hunk 1]
```diff
@@ -10,15 +10,15 @@ These are instructions on how to make a release.
 
 ## Steps to make a release
 
-1. Create a PR updating `jupyterlab-server-proxy/yarl.lock` and continue only
+1. Create a PR updating `labextension/yarl.lock` and continue only
```
Comment (manics) [inline RELEASE.md:13]:
```suggestion
1. Create a PR updating `labextension/yarn.lock` and continue only
```

**Comment by consideRatio** [comment]
> How old is JupyterLab 2? Given this is a major release can we drop it?

Maybe, but I'd like to uncouple that discussion from this PR. In my mind, we should if several other projects stop supporting lab2 also.

**Comment by consideRatio** [comment]
@manics I rebased to resolve a merge conflict. No changes was made besides that in the force push.

**Comment by manics** [comment]
OK! I'll leave it to @bollwyvl to check and merge.

**Comment by bollwyvl** [comment]
Sorry I missed this earlier. The changes look fine, and I'm +1 on merging it as-is, with perhaps some iteration on (partial) ideas on #370 before doing a full release.

**Review by bollwyvl — APPROVED** [review]

**Comment by consideRatio** [comment]
Thanks @bollwyvl and @manics! I self-merged this given its approved by you both now

**Pull request merged**